src/domain/home/form/HomePreliminaryForm.tsx

```diff
@@ -42,17 +42,24 @@ const HomePreliminaryForm: FunctionComponent<Props> = ({
       ...defaultFormValues,
       ...{
         child: {
+          // Ensure date that saved in redux store was using backend time format:
           birthdate: formatTime(
             newMoment(
               `${values.child.birthdate.year}-${values.child.birthdate.month}-${values.child.birthdate.day}`,
               'YYYY-MM-DD'
             )
           ),
-          // Ensure date that saved in redux store was using backend time format.
           firstName: stateFormValues.child.firstName,
-          lastName: stateFormValues.child.firstName,
+          lastName: stateFormValues.child.lastName,
           homeCity: values.child.homeCity,
         },
+        guardian: {
+          firstName: stateFormValues.guardian.firstName,
```

Comment:
You sure this is needed? It suppose to still work without changes on guardian

Reply:
If you change guardian info in the registration form, and then go back to the front page form, and then back again your own changes will reverted back to what tunnistamo has

Reply:
Then we have to find another way to update the state, those changes are non-necessary too long

Reply:
Any suggestions?

Reply:
Done